algopy_testing/examples/auction/test_contract.py

```diff
@@ -0,0 +1,143 @@
+from collections.abc import Generator
+from typing import Any
+from unittest.mock import MagicMock
+
+import pytest
+from algopy import UInt64, gtxn
+from algopy_testing import TestContext, blockchain_context
+
+from .contract import AuctionContract
+
+
+@pytest.fixture()
+def context() -> Generator[TestContext[Any], None, None]:
+    with blockchain_context() as ctx:
+        yield ctx
+
+
+def test_opt_into_asset(
+    context: TestContext[Any],
+) -> None:
+    # Arrange
+    dummy_account = context.any_account()
+    context.set_global_fields(
+        {
+            "creator_address": dummy_account,
+            "current_application_address": dummy_account,
```

Comment:
minor: this test should be using a different account for the application address